src/Compilers/CSharp/Test/Semantic/Semantics/RefEscapingTests.cs

```diff
@@ -4313,6 +4313,180 @@ static void Main()
+        }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/75802")]
+        public void CollectionInitializer_UnscopedRef_Assignment()
+        {
+            var source = """
+                using System.Collections;
+                using System.Diagnostics.CodeAnalysis;
+
+                class C
+                {
+                    R M()
+                    {
+                        var local = 1;
+                        var r = new R() { local };
+                        return r;
+                    }
+                }
+
+                ref struct R : IEnumerable
+                {
+                    public void Add([UnscopedRef] in int x) { }
+                    IEnumerator IEnumerable.GetEnumerator() => throw null;
+                }
+                """;
+            CreateCompilation([source, UnscopedRefAttributeDefinition]).VerifyDiagnostics(
+                // (10,16): error CS8352: Cannot use variable 'r' in this context because it may expose referenced variables outside of their declaration scope
+                //         return r;
+                Diagnostic(ErrorCode.ERR_EscapeVariable, "r").WithArguments("r").WithLocation(10, 16));
+        }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/75802")]
+        public void CollectionInitializer_ScopedRef_Return()
+        {
+            var source = """
+                using System.Collections;
+
+                class C
+                {
+                    R M()
+                    {
+                        var local = 1;
+                        return new R() { local };
+                    }
+                }
+
+                ref struct R : IEnumerable
+                {
+                    public void Add(in int x) { }
+                    IEnumerator IEnumerable.GetEnumerator() => throw null;
+                }
+                """;
+            CreateCompilation(source).VerifyDiagnostics();
+        }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/75802")]
+        public void CollectionInitializer_ScopedRef_Assignment()
+        {
+            var source = """
+                using System.Collections;
+
+                class C
+                {
+                    R M()
+                    {
+                        var local = 1;
+                        var r = new R() { local };
+                        return r;
+                    }
+                }
+
+                ref struct R : IEnumerable
+                {
+                    public void Add(in int x) { }
+                    IEnumerator IEnumerable.GetEnumerator() => throw null;
+                }
+                """;
+            CreateCompilation(source).VerifyDiagnostics();
+        }
+
+        [Theory, WorkItem("https://github.com/dotnet/roslyn/issues/75802")]
+        [InlineData("[System.Diagnostics.CodeAnalysis.UnscopedRef]")]
+        [InlineData("")]
```

Comment:
>[InlineData("")] [](http://example.com/codeflow?start=8&length=16)

It looks like the error reported when there is no `[UnscopedRef]` attribute may be unrelated to this change. Instead, it looks like `RefSafetyAnalysis.HasLocalScope` is not expecting a `ref struct` to implement `IEnumerable`, other than in error cases.

Reply:
Similarly to `CollectionExpression_Builder`, I was just verifying that references cannot be captured with collection expressions. Haven't realized the error is unexpected. Should we file an issue for that?

Reply:
> Haven't realized the error is unexpected. Should we file an issue for that?

Yes, we should file an issue for the error reported when the `Add(in int x)` parameter is not marked `[UnscopedRef]`.